File: AIOrchestraUI/src/app/models/base.response.ts
class ErrorDetail {
    code: string;
    message: string;
    details: string;

    constructor(code: string, message: string, details: string) {
        this.code = code;
        this.message = message;
        this.details = details;
    }
}

export class BaseResponse<T> {
    isSuccess: boolean;
    isFailure: boolean;
    statusCode: number;
    error: ErrorDetail;
    processingTime: number;
    servicedBy: number;
    additionalDetails: any;
    operationId: string;
    apiVersion: string;
    value: T;
    handlerMethod: string;

    constructor(
        isSuccess: boolean,
        isFailure: boolean,
        statusCode: number,
        error: ErrorDetail,
        processingTime: number,
        servicedBy: number,
        additionalDetails: any,
        operationId: string,
        apiVersion: string,
        value: T,
        handlerMethod: string
    ) {
        this.isSuccess = isSuccess;
        this.isFailure = isFailure;
        this.statusCode = statusCode;
        this.error = new ErrorDetail(error.code, error.message, error.details);
        this.processingTime = processingTime;
        this.servicedBy = servicedBy;
        this.additionalDetails = additionalDetails;
        this.operationId = operationId;
        this.apiVersion = apiVersion;
        this.value = value;
        this.handlerMethod = handlerMethod;
    }
}

File: MusicRecommenderService/contracts/baseResponse.py
from typing import Dict, Any
from contracts.baseContract import BaseContract

class BaseResponse(BaseContract):
    def __init__(self):
        super().__init__()
        self.is_success: bool = True
        self.is_failure: bool = False
        self.status_code: int = 200
        self.error: Error = Error()
        self.processing_time: int = 0
        self.serviced_by: Any = None
        self.additional_details: Dict[str, str] = {}
    
    def to_dict(self):
        parent_dict = super().to_dict()
        return {
            **parent_dict,
            "IsSuccess": self.is_success,
            "IsFailure": self.is_failure,
            "StatusCode": self.status_code,
            "Error": self.error.__dict__,
            "ProcessingTime": self.processing_time,
            "ServicedBy": self.serviced_by,
            "AdditionalDetails": self.additional_details
        }

class Error:
    def __init__(self):
        self.code: str = ""
        self.message: str = ""
        self.details: str = ""